app/controllers/users_controller.rb

```diff
@@ -1,6 +1,16 @@
 class UsersController < ApplicationController
+  before_action :admin_user, only: %i(destroy)
+  before_action :correct_user, only: %i(edit update)
+  before_action :logged_in_user, only: %i(index edit update)
+  before_action :find_user, only: %i(show update edit destroy)
```

Comment:
cho line này đứng trc line 2 để tái sử dụng việc xử lý tìm kiếm user

```suggestion
  before_action :find_user, except: %i(new create index)
```